File: kanban-board-reactive-api/src/main/java/pl/edu/utp/kanbanboard/websocket/service/ChatService.java
package pl.edu.utp.kanbanboard.websocket.service;

import pl.edu.utp.kanbanboard.websocket.model.Message;
import reactor.core.publisher.Flux;

public interface ChatService {
    Flux<Message> getHistory(String projectId);

    void addMessageToHistory(String projectId, Message message);
}

File: kanban-board-frontend/src/app/tasks/chat/chat.service.ts
import {Injectable} from '@angular/core';
import {environment} from '../../../environments/environment';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Message} from './message';
import {AuthService} from '../../auth/auth-service/auth.service';

@Injectable({
  providedIn: 'root'
})
export class ChatService {
  private CHAT_URL = environment.backendUrl + '/api/chat';
  private HEADERS = this.authService.getAuthHeaders();

  constructor(private httpClient: HttpClient,
              private authService: AuthService) { }

  getHistory(projectId: string): Observable<Message[]> {
    return this.httpClient.get<Message[]>(this.CHAT_URL + '/history/' + projectId, {headers: this.HEADERS});
  }
}
